asciidoc-extensions/src/main/resources/ExampleNumberingProcessor.rb

```diff
@@ -0,0 +1,43 @@
+# encoding: UTF-8
+require 'asciidoctor'
+require 'asciidoctor/extensions'
+
+class ExampleIndexes
+  attr_accessor :sectionIndex
+  @sectionIndex
+  attr_accessor :exampleIndex
+  @exampleIndex
+  attr_accessor :tableIndex
+  @tableIndex
+  def initialize(sect)
+    @sectionIndex = sect
+    @tableIndex = 1
+    @exampleIndex = 1
+  end
+end
+
+class ExampleNumberingProcessor < Asciidoctor::Extensions::Treeprocessor
```

Comment:
Out of curiosity, what was the issue when writing it in Java? It'd be nicer to not introduce another language, but we can do it if there is no other way.